### PR TITLE
feat(agents): add validation for unique sub-agent names (#3557)

### PR DESCRIPTION
**Pull request by sarojrout**
**Please ensure you have read the [contribution guide](https://github.com/google/adk-python/blob/main/CONTRIBUTING.md) before creating a pull request.**

### Link to Issue or Description of Change

**1. Link to an existing issue (if applicable):**

- Closes: #3557
- Related: #_issue_number_

**2. Or, if no issue exists, describe the change:**

_If applicable, please follow the issue templates to provide as much detail as
possible._

**Problem:**
When creating a BaseAgent with multiple sub-agents, there was no validation to ensure that all sub-agents have unique names. This could lead to confusion when trying to find or reference specific sub-agents by name, as duplicate names would make it ambiguous which agent is being referenced.
**Solution:**
Added a @field_validator for the sub_agents field in BaseAgent that validates all sub-agents have unique names. The validator:
Checks for duplicate names in the sub-agents list
Raises a ValueError with a clear error message listing all duplicate names found
Returns the validated list if all names are unique
Handles edge cases like empty lists gracefully

### Testing Plan

_Please describe the tests that you ran to verify your changes. This is required
for all PRs that are not small documentation or typo fixes._

**Unit Tests:**

- [x] I have added or updated unit tests for my change.
- [x] All unit tests pass locally.

_Please include a summary of passed `pytest` results._
Added 4 new test cases in tests/unittests/agents/test_base_agent.py:
test_validate_sub_agents_unique_names_single_duplicate: Verifies that a single duplicate name raises ValueError
test_validate_sub_agents_unique_names_multiple_duplicates: Verifies that multiple duplicate names are all reported in the error message
test_validate_sub_agents_unique_names_no_duplicates: Verifies that unique names pass validation successfully
test_validate_sub_agents_unique_names_empty_list: Verifies that empty sub-agents list passes validation
All tests pass locally. You can run with:
pytest tests/unittests/agents/test_base_agent.py::test_validate_sub_agents_unique_names_single_duplicate tests/unittests/agents/test_base_agent.py::test_validate_sub_agents_unique_names_multiple_duplicates tests/unittests/agents/test_base_agent.py::test_validate_sub_agents_unique_names_no_duplicates tests/unittests/agents/test_base_agent.py::test_validate_sub_agents_unique_names_empty_list -v
**Manual End-to-End (E2E) Tests:**

_Please provide instructions on how to manually test your changes, including any
necessary setup or configuration. Please provide logs or screenshots to help
reviewers better understand the fix._

Test Case 1: Duplicate names should raise error
from google.adk.agents import Agent

agent1 = Agent(name="sub_agent", model="gemini-2.5-flash")
agent2 = Agent(name="sub_agent", model="gemini-2.5-flash")  # Same name

# This should raise ValueError
try:
    parent = Agent(
        name="parent",
        model="gemini-2.5-flash",
        sub_agents=[agent1, agent2]
    )
except ValueError as e:
    print(f"Expected error: {e}")
    # Output: Found duplicate sub-agent names: `sub_agent`. All sub-agents must have unique names.

Test Case 2: Unique names should work
from google.adk.agents import Agent

agent1 = Agent(name="agent1", model="gemini-2.5-flash")
agent2 = Agent(name="agent2", model="gemini-2.5-flash")

# This should work without error
parent = Agent(
    name="parent",
    model="gemini-2.5-flash",
    sub_agents=[agent1, agent2]
)
print("Success: Unique names validated correctly")

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/google/adk-python/blob/main/CONTRIBUTING.md) document.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have manually tested my changes end-to-end.
- [x] Any dependent changes have been merged and published in downstream modules.

### Additional context

This change adds validation at the BaseAgent level, so it automatically applies to all agent types that inherit from BaseAgent (e.g., LlmAgent, LoopAgent, etc.). The validation uses Pydantic's field validator system, which runs during object initialization, ensuring the constraint is enforced early and consistently.
The error message clearly identifies which names are duplicated, making it easy for developers to fix the issue:

### CODE REVIEW

File: src/google/adk/agents/base_agent.py [hunk 1]
```diff
@@ -563,6 +563,44 @@ def validate_name(cls, value: str):
...
+      The validated list of sub-agents.
+
+    Raises:
+      ValueError: If duplicate sub-agent names are found.
+    """
+    if not value:
+      return value
+
+    seen_names: set[str] = set()
+    duplicates: list[str] = []
```
Comment (slilichenko) [inline src/google/adk/agents/base_agent.py]: Minor nit - probably needs to be a set rather than a list.

File: src/google/adk/agents/base_agent.py [hunk 2]
```diff
@@ -563,6 +563,44 @@ def validate_name(cls, value: str):
...
+    for sub_agent in value:
+      name = sub_agent.name
+      if name in seen_names:
+        duplicates.append(name)
+      else:
+        seen_names.add(name)
+
+    if duplicates:
+      duplicate_names_str = ', '.join(
+          f'`{name}`' for name in sorted(set(duplicates))
```
Comment (slilichenko) [inline src/google/adk/agents/base_agent.py]: If you had a set to start with, no need to do this collection copy operation.

Comment (sarojrout) [reply src/google/adk/agents/base_agent.py]: you are absolute right @slilichenko . will fix it with set instead of list. i overlooked. thanks!

File: src/google/adk/agents/base_agent.py (hunk 1 above)
Comment (sarojrout) [reply src/google/adk/agents/base_agent.py]: done @slilichenko . pls review and merge if looks good. thanks!

**Comment by rohityan** [comment]
Hi @sarojrout , can you fix the lint errors.

**Comment by sarojrout** [comment]
> Hi @sarojrout , can you fix the lint errors.

done. can you pls verify? thanks!

**Comment by rohityan** [comment]
Hi @slilichenko, can you please review this now.

**Comment by slilichenko** [comment]
@ryanaiagent LGTM, but I am not an official reviewer/committer on this project.

**Comment by rohityan** [comment]
Hi @DeanChensj , can you please review this.
